src/core/ir/resolver_context_like.rs

```diff
@@ -68,3 +69,87 @@ impl<'a> ResolverContextLike for ResolverContext<'a> {
         self.inner.ctx.add_error(error)
     }
 }
+
+#[derive(Debug)]
+pub struct SelectionField {
+    name: String,
+    args: Vec<(String, String)>,
+    directive: async_graphql::ServerResult<Vec<ConstDirective>>,
```

Comment:
```suggestion
    directives: async_graphql::ServerResult<Vec<ConstDirective>>,
```

Reply:
we don't use ServerResult anyway after that. Consider storing just vec or `Option<Vec>`